packages/timeline/src/ReactGantt.ts

```diff
@@ -0,0 +1,467 @@
+        d.props.width = d.props.width < 1 ? 1 : d.props.width;
+        d.props.height = d.props.height < 1 ? 1 : d.props.height;
+
+        let text = this.truncateText(d.props.text, d.props.width - padding, this._maxFontScale);
+
+        if(text !== d.props.text){
+            text = this.truncateText(d.props.text, d.props.width - padding);
+        } else {
+            d.props.fontSize = this._maxFontScale * options.fontSize;
+        }
+
+        render(
+            this._rangeRenderer,
+            {
+                ...options,
+                ...d.props,
+                text,
+            },
+            d.that
+        );
+    }
+
+    setRangeOptions(){
+        this._rangeOptions = {
+            rangePadding: this.rangePadding(),
+            fontFamily: this.fontFamily(),
+            fontSize: this.fontSize(),
+            strokeWidth: this.strokeWidth(),
+            fill: this.fill(),
+            stroke: this.stroke(),
+            textFill: this.rangeFontColor(),
+            cornerRadius: this.cornerRadius(),
+        }
+    }
+    
+    public _transform = {k:1,x:0,y:0};
+    zoomed(transform) {
+        this._transform = transform;
+        console.log('transform', transform);
+        switch(this.renderMode()){
+            case "scale-all": 
+                this._zoomScale = transform.k;
+                this._zoomTranslate = [transform.x, 0];
+                this._zoomG.attr("transform", `translate(${transform.x} ${0})scale(${transform.k} 1)`);
+                break;
+            default:
+                const options = this._rangeOptions;
+                this.data().forEach((d,i)=>{
+                    this.renderRangeElement(d, i, true, options);
+                });
+        }
+
+        this.zoomedHook(transform);
+    }
+
+    zoomedHook(transform) {
+
+    }
+
+    private _maxBucket: number = 0;
+    private _minRangeWidth: number = Infinity;
+    calcBuckets(startKey: string | number, endKey: string | number) {
+        const bucketMap = {};
+        const tol = this.overlapTolerence();
+        const buckets = [{end:-Infinity}];
+        this.data().forEach((d,i)=>{
+            const rangeWidth = d[endKey] - d[startKey];
+            if(this._minRangeWidth > rangeWidth)this._minRangeWidth = rangeWidth;
+            for (let i2 = 0; i2 < buckets.length; ++i2) {
+                if (i === 0 || buckets[i2][endKey] + tol <= d[startKey]) {
+                    bucketMap[i] = i2;
+                    if(this._maxBucket < i2)this._maxBucket = i2;
+                    buckets[i2][endKey] = d[endKey];
+                    break;
+                }
+            }
+            if(bucketMap[i] === undefined){
+                bucketMap[i] = buckets.length;
+                const b = {};
+                b[endKey] = d[endKey];
+                buckets.push(b as any);
+            }
+            
+            if(this._maxBucket < bucketMap[i])this._maxBucket = bucketMap[i];
+        });
+        return bucketMap;
```

Comment:
Could this code be commoned up with the existing version (and any other bits)?

Reply:
It could, but I imagine the bottom (range) portion of the existing version will eventually be replaced with this one.

Reply:
Can you open a ticket for that?

Reply:
Opened: https://github.com/hpcc-systems/Visualization/issues/3683